src/mbgl/renderer/layers/render_symbol_layer.cpp

```diff
@@ -670,17 +654,31 @@ style::TextPaintProperties::PossiblyEvaluated RenderSymbolLayer::textPaintProper
 
 void RenderSymbolLayer::setRenderTiles(RenderTiles tiles, const TransformState& state) {
     renderTiles = std::move(tiles);
-    // Sort symbol tiles in opposite y position, so tiles with overlapping symbols are drawn
-    // on top of each other, with lower symbols being drawn on top of higher symbols.
-    std::sort(renderTiles.begin(), renderTiles.end(), [&state](const auto& a, const auto& b) {
-        Point<float> pa(a.get().id.canonical.x, a.get().id.canonical.y);
-        Point<float> pb(b.get().id.canonical.x, b.get().id.canonical.y);
+    const auto comp = [bearing = state.getBearing()](const RenderTile& a, const RenderTile& b) {
+        Point<float> pa(a.id.canonical.x, a.id.canonical.y);
+        Point<float> pb(b.id.canonical.x, b.id.canonical.y);
 
-        auto par = util::rotate(pa, state.getBearing());
-        auto pbr = util::rotate(pb, state.getBearing());
+        auto par = util::rotate(pa, bearing);
+        auto pbr = util::rotate(pb, bearing);
 
-        return std::tie(b.get().id.canonical.z, par.y, par.x) < std::tie(a.get().id.canonical.z, pbr.y, pbr.x);
-    });
+        return std::tie(b.id.canonical.z, par.y, par.x) < std::tie(a.id.canonical.z, pbr.y, pbr.x);
+    };
+    // Sort symbol tiles in opposite y position, so tiles with overlapping symbols are drawn
+    // on top of each other, with lower symbols being drawn on top of higher symbols.
+    std::sort(renderTiles.begin(), renderTiles.end(), comp);
+
+    placementData.clear();
```

Comment:
It would be useful to add comment here, and related to:
`
cross-tile indexing and placement operations take less time as the required symbol bucket are prepared beforehand in the RenderSymbolLayer.
`
e.g. 
We fill the container here to speedup the placement operations at...

 I find this comment useful because both are currently invoked from Renderer::Impl::render.

Reply:
This code will be further refactored in the follow-up patches.